packages/trpc/server/routers/loggedInViewer/outOfOffice.handler.ts

```diff
@@ -4,11 +4,48 @@ import dayjs from "@calcom/dayjs";
 import { sendBookingRedirectNotification } from "@calcom/emails";
 import { getTranslation } from "@calcom/lib/server";
 import prisma from "@calcom/prisma";
+import { MembershipRole } from "@calcom/prisma/enums";
 import type { TrpcSessionUser } from "@calcom/trpc/server/trpc";
 
 import { TRPCError } from "@trpc/server";
 
-import type { TOutOfOfficeDelete, TOutOfOfficeInputSchema } from "./outOfOffice.schema";
+import type {
+  TOutOfOfficeDelete,
+  TOutOfOfficeEntriesListSchema,
+  TOutOfOfficeInputSchema,
+} from "./outOfOffice.schema";
+
+// function getTeam() checks if there is a team where 'adminUserId' is admin or owner
+// and 'memberUserId' is a member.
+// If exists returns the first found such team.
+const getTeam = async (adminUserId: number, memberUserId: number) => {
+  return await prisma.team.findFirst({
+    where: {
+      AND: [
+        {
+          members: {
+            some: {
+              userId: adminUserId,
+              role: {
+                in: [MembershipRole.ADMIN, MembershipRole.OWNER],
+              },
+            },
+          },
+        },
+        {
+          members: {
+            some: {
+              userId: memberUserId,
+            },
+          },
+        },
+      ],
+    },
```

Comment:
Too many changes in the backend inside a single file. We definitely need to add tests